### ISSUE TITLE
Build failure 'boost/circular_buffer.hpp' file not found 

### ISSUE THREAD

**Issue by Dorwido** [mapsme/omim#10250]
Trying to build the android version leads to that error.

Uploaded a scan of the build try:

https://scans.gradle.com/s/x2tknafzxwbai

**Comment by gyunaev** [comment]
Do you have the file 3party/boost/boost/circular_buffer.hpp?

Did you retrieve the whole repo with submodules?

**Comment by Dorwido** [comment]
Yes the file does exist after it didnt work, I even tried to install the ubuntu dev package version, to see if that helps which don't do either.
All submodules installed.

Compiling the desktop version works btw.

`locate -b circular_buffer.hpp
/home/tsanders/projekte/private/omim/3party/boost/boost/circular_buffer.hpp
/home/tsanders/projekte/private/omim/3party/boost/boost/ptr_container/ptr_circular_buffer.hpp
/home/tsanders/projekte/private/omim/3party/boost/boost/ptr_container/serialize_ptr_circular_buffer.hpp
/usr/include/boost/circular_buffer.hpp
/usr/include/boost/ptr_container/ptr_circular_buffer.hpp
/usr/include/boost/ptr_container/serialize_ptr_circular_buffer.hpp
`

**Comment by gyunaev** [comment]
Ubuntu packages are not used when you compile for Android.

**Comment by Dorwido** [comment]
Found now a solution to let it compile, in the CMakeLists.txt

`find_package(Boost 1.64)
if (NOT Boost_FOUND)
  if (DEFINED ENV{BOOST_ROOT})
    if (DARWIN)
      set(Boost_INCLUDE_DIR "$ENV{BOOST_ROOT}/include")
      set(Boost_LIBRARY_DIR "$ENV{BOOST_ROOT}/lib")
    else()
      set(Boost_INCLUDE_DIR "$ENV{BOOST_ROOT}")
      set(Boost_LIBRARY_DIR "$ENV{BOOST_ROOT}/libs")
    endif()
  else()
    set(Boost_INCLUDE_DIR "${OMIM_ROOT}/3party/boost")
    set(Boost_LIBRARY_DIR "${OMIM_ROOT}/3party/boost/libs")
  endif()
  find_package(Boost 1.64)
endif()`

The first find_package(Boost 1.64) is triggered as true with finding the system installed version:
Boost version: 1.65.1

adding a:
`message("Boost include dir: " ${Boost_INCLUDE_DIR})
message("Boost library dir: " ${Boost_LIBRARY_DIR})`

after it prints:

```
Boost include dir: /usr/include
Boost library dir: 
```
Changing the find to:
` find_package(Boost 1.64 EXACT)`

then the print is:
```
Boost include dir: /home/tsanders/projekte/private/omim/3party/boost
Boost library dir: /home/tsanders/projekte/private/omim/3party/boost/libs
```
And compiling works without a problem.

But I guess the EXACT is not a real solution, because if the OS installed one would be also 1.64 , the error would the same one.
